tb_plugin/test/test_tensorboard_end2end.py

```diff
@@ -17,24 +23,44 @@ class TestEnd2End(unittest.TestCase):
     #    self._test_tensorboard_with_arguments(test_folder, expected_runs, {'TORCH_PROFILER_START_METHOD':'spawn'})
 
     def test_tensorboard_end2end(self):
-        test_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)),'../samples')
+        test_folder = get_samples_dir()
         expected_runs = b'["resnet50_num_workers_0", "resnet50_num_workers_4"]'
         
         print("starting spawn mode testing...")
         self._test_tensorboard_with_arguments(test_folder, expected_runs, {'TORCH_PROFILER_START_METHOD':'spawn'})
 
     def test_tensorboard_fork(self):
-        test_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)),'../samples')
+        test_folder = get_samples_dir()
         expected_runs = b'["resnet50_num_workers_0", "resnet50_num_workers_4"]'
 
         print("starting fork mode testing")
         self._test_tensorboard_with_arguments(test_folder, expected_runs)
 
     def test_tensorboard_with_path_prefix(self):
-        test_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)),'../samples')
+        test_folder = get_samples_dir()
         expected_runs = b'["resnet50_num_workers_0", "resnet50_num_workers_4"]'
         self._test_tensorboard_with_arguments(test_folder, expected_runs, path_prefix='/tensorboard/viewer/')
 
+    def test_tensorboard_with_symlinks(self):
+        logdir = tempfile.mkdtemp(prefix="tensorboard_logdir")
+
+        samples_dir = get_samples_dir()
+
+        # Desired layout:
+        # logdir/
+        #     run_A/ --> path/to/samples/resnet50_num_workers_0/
+        #     run_B/ --> path/to/samples/resnet50_num_workers_4/
+        for target, link_name in [
+            ("resnet50_num_workers_0", "run_A"),
+            ("resnet50_num_workers_4", "run_B"),
+        ]:
+            os.symlink(os.path.join(samples_dir, target), os.path.join(logdir, link_name))
```

Comment:
Do we need mix both symlinks and ordinary files together to test both the mix cases?

Reply:
BTW, would you please create an issue to track this change?

Reply:
Done for both! https://github.com/pytorch/kineto/issues/331